dxr/plugins/python/indexers.py

```diff
@@ -153,30 +227,35 @@ def __init__(self, path, contents, plugin_name, tree, python_path,
         self.derived_classes = derived_classes
         self.names = names
 
+        self.module_name = path_to_module(self.python_path, self.path)
+
     def is_interesting(self):
         return is_interesting(self.path)
 
     def needles_by_line(self):
+        self.analyze_tokens()
+
+        visitor = IndexingNodeVisitor(self)
+        syntax_tree = ast.parse(self.contents.encode('utf-8'))
+
         return iterable_per_line(
             with_start_and_end(
                 split_into_lines(
-                    self._all_needles()
+                    visitor.get_needles(syntax_tree)
                 )
             )
         )
 
-    def _all_needles(self):
-        """Return an iterable of needles in (needle name, value, Extent)
-        format.
+    def analyze_tokens(self):
+        """Split the file into tokens and analyze them for data needed
+        for indexing.
 
         """
-        module_name = path_to_module(self.python_path, self.path)
-
         # AST nodes for classes and functions point to the position of
         # their 'def' and 'class' tokens. To get the position of their
         # names, we look for 'def' and 'class' tokens and store the
         # position of the token immediately following them.
-        node_start_table = {}
+        self.node_start_table = {}
         previous_start = None
         token_gen = tokenize.generate_tokens(StringIO(self.contents).readline)
```

Comment:
We're passing this a unicode and the visitor a str. Is that going to cause inconsistencies?